bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/scanner/RowMatcher.java

```diff
@@ -0,0 +1,342 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.scanner;
+
+import com.google.bigtable.v1.Cell;
+import com.google.bigtable.v1.Column;
+import com.google.bigtable.v1.Family;
+import com.google.bigtable.v1.Row;
+import com.google.protobuf.ByteString;
+
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@code RowMatcher} provides a convenient Matcher-based language for evaluating
+ * {@link Row} test conditions.  Example usage:
+ *
+ *   import static com.google.cloud.bigtable.grpc.scanner.RowMatcher.matchesRow;
+ *
+ *   assertThat(row,
+ *       matchesRow("row key").withFamily("Family").withColumn("column").withAnyCell());
+ */
+public class RowMatcher extends TypeSafeMatcher<Row> {
+  /**
+   * Creates a new row matcher using the provided matcher to evaluate the
+   * row key.
+   *
+   * @param rowKey Matcher to use for the row key.
+   * @return A new {@code RowMatcher} object.
+   */
+  public static RowMatcher matchesRow(Matcher<ByteString> rowKey) {
+    return new RowMatcher(rowKey);
+  }
+
+  /**
+   * A convenience method for constructing a new row matcher with key equal
+   * to the UTF-8 bytes for the given string.
+   *
+   * @see #matchesRow(Matcher)
+   */
+  public static RowMatcher matchesRow(String rowKey) {
+    return new RowMatcher(CoreMatchers.equalTo(ByteString.copyFromUtf8(rowKey)));
+  }
+
+  /**
+   * Adds a new column family with the given name to the expected value for
+   * this row.  Subsequent {@link #withColumn(Matcher)} calls will be added
+   * to the most recently added family.
+   *
+   * Usage:
+   *   {@code matchesRow("key").withFamily("family")}
+   */
+  public RowMatcher withFamily(String familyName) {
+    lastFamily = new FamilyMatcher(familyName);
+    families.put(familyName, lastFamily);
+    return this;
+  }
+
+  /**
+   * Adds a column with qualifier matching the given matcher to the current
+   * column family.  A column family <em>must</em> have already been added
+   * to the expected row with {@link #withFamily(String)} prior to calling
+   * {@code withColumn}.
+   *
+   * Example:
+   *   {@code withFamily("family").withColumn(any(ByteString.class))}
+   */
+  public RowMatcher withColumn(Matcher<ByteString> columnName) {
+    lastColumn = new ColumnMatcher(columnName);
```

Comment:
hm.  It's interesting to be able to do withColumn(..).withCell(..), but it seems a tad magical.  Perhaps withColumn(name, cellValue)?